packages/backend/src/core/TransactionStatusMonitor.ts

```diff
@@ -0,0 +1,30 @@
+import { TransactionStatusService } from './TransactionStatusService'
+
+const MINUTE = 1000 * 60
+
+export class TransactionStatusMonitor {
+  private timeout?: NodeJS.Timeout
+
+  constructor(
+    private readonly transactionStatusService: TransactionStatusService,
+    private readonly syncInterval = MINUTE
+  ) {}
+
+  private scheduleNextCheck() {
+    this.timeout = setTimeout(async () => {
+      await this.transactionStatusService.syncTransactions()
+      this.scheduleNextCheck()
+    }, this.syncInterval)
+  }
+
+  start() {
+    this.scheduleNextCheck()
+  }
+
+  stop() {
+    if (!this.timeout) {
+      return
+    }
+    clearTimeout(this.timeout)
```

Comment:
This doesn't work if we're in the middle of execution. An easier way is to have a member `running: boolean` and `scheduleNextCheck` just checks this and the callback in setTimeout does too